### PR TITLE
erlang 25.0.3

### PR DESCRIPTION
**Pull request by fishtreesugar**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by carlocab** [comment]
`couchdb` and `syntaxerl` need revision bumps.

**Comment by cho-m** [comment]
Odd behavior over the last few patch versions:
- Error in **25.0.1** - #103312
- No error in **25.0.2** - #104123
- Error in **25.0.3**

Interesting that all CI runners exhibit the same behavior, so less likely to be non-deterministic, though could try locally to confirm. I wonder if **25.0.4** will pass giving an even/odd trend.

**Comment by cho-m** [comment]
Added some workarounds to try. Not sure if they will actually resolve issue since I don't know root cause.

For now, trying to avoid using rebar (2) or old copies rebar3 with Erlang 25.

File: Formula/couchdb.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class Couchdb < Formula
   mirror "https://archive.apache.org/dist/couchdb/source/3.2.2/apache-couchdb-3.2.2.tar.gz"
   sha256 "69c9fd6f80133557f68a02e92dda72a4fd646d646f429f45bb8329a30f82f20e"
   license "Apache-2.0"
+  revision 1
```
Comment (cho-m) [inline Formula/couchdb.rb:8]: If preferred, we could do the modification in a separate PR and avoid rebuilding bottles as `erlang` is a build-only dependency

Comment (cho-m) [reply Formula/couchdb.rb:8]: Cherry-picked into #106767. Can drop commits once that PR is in.

**Comment by carlocab** [comment]
On Monterey:
```
  could not write file '../specs/erl_prim_loader.specs': no such file or directory
  make[2]: *** [../specs/specs_erl_prim_loader.xml] Error 2
  make[2]: *** Waiting for unfinished jobs....
  could not write file '../specs/erl_tracer.specs': no such file or directory
  make[2]: *** [../specs/specs_erl_tracer.xml] Error 2
  could not write file '../specs/init.specs': no such file or directory
  make[2]: *** [../specs/specs_init.xml] Error 2
  could not write file '../specs/persistent_term.specs': no such file or directory
  make[2]: *** [../specs/specs_persistent_term.xml] Error 2
  could not write file '../specs/erlang.specs': no such file or directory
  make[2]: *** [../specs/specs_erlang.xml] Error 2
  make[1]: *** [docs] Error 2
  make: *** [docs] Error 2
```
It doesn't seem like anything changed since we last built it. Race condition maybe?

**Review by cho-m — APPROVED** [review]
